cluster/aws/util.sh

```diff
@@ -1242,70 +1360,104 @@ function start-minions() {
 
   write-node-env
 
-  (
-    # We pipe this to the ami as a startup script in the user-data field.  Requires a compatible ami
-    echo "#! /bin/bash"
-    echo "mkdir -p /var/cache/kubernetes-install"
-    echo "cd /var/cache/kubernetes-install"
-    echo "cat > kube_env.yaml << __EOF_KUBE_ENV_YAML"
-    cat ${KUBE_TEMP}/node-kube-env.yaml
-    echo "AUTO_UPGRADE: 'true'"
-    # TODO: get rid of these exceptions / harmonize with common or GCE
-    echo "DOCKER_STORAGE: $(yaml-quote ${DOCKER_STORAGE:-})"
-    echo "API_SERVERS: $(yaml-quote ${MASTER_INTERNAL_IP:-})"
-    echo "__EOF_KUBE_ENV_YAML"
-    echo ""
-    echo "wget -O bootstrap ${BOOTSTRAP_SCRIPT_URL}"
-    echo "chmod +x bootstrap"
-    echo "mkdir -p /etc/kubernetes"
-    echo "mv kube_env.yaml /etc/kubernetes"
-    echo "mv bootstrap /etc/kubernetes/"
-    echo "cat > /etc/rc.local << EOF_RC_LOCAL"
-    echo "#!/bin/sh -e"
-    # We want to be sure that we don't pass an argument to bootstrap
-    echo "/etc/kubernetes/bootstrap"
-    echo "exit 0"
-    echo "EOF_RC_LOCAL"
-    echo "/etc/kubernetes/bootstrap"
-  ) > "${KUBE_TEMP}/node-user-data"
-
-  # Compress the data to fit under the 16KB limit (cloud-init accepts compressed data)
-  gzip "${KUBE_TEMP}/node-user-data"
-
-  local public_ip_option
-  if [[ "${ENABLE_NODE_PUBLIC_IP}" == "true" ]]; then
-    public_ip_option="--associate-public-ip-address"
-  else
-    public_ip_option="--no-associate-public-ip-address"
-  fi
-  local spot_price_option
-  if [[ -n "${NODE_SPOT_PRICE:-}" ]]; then
-    spot_price_option="--spot-price ${NODE_SPOT_PRICE}"
+  if [[ "${PRESET_INSTANCES_ENABLED:-}" == "yes" ]]; then
```

Comment:
Many duplicates in this block for PRESET_INSTANCES_ENABLED = yes/no, any chance those can be consolidated?

Reply:
Updated